### PR TITLE
Refactor EventHandlerTagHelperDescriptorProvider

### PR DESCRIPTION
**Pull request by bytedreamer**
Learning how Blazor compiles, did some refactoring along the way

### CODE REVIEW

**Comment by rynowak** [comment]
Hi, is there any issue that this is associated with? What problem is being solved?

**Comment by bytedreamer** [comment]
No issues, did some refactoring while learning how the code works.

**Comment by bytedreamer** [comment]
@rynowak I see all of the Blazor compiler code has been deleted under Microsoft.AspNetCore.Components.Razor.Extensions. Has it been migrated to another location?

File: src/Components/src/Microsoft.AspNetCore.Components.Razor.Extensions/EventHandlerTagHelperDescriptorProvider.cs [hunk 1]
```diff
@@ -124,9 +115,10 @@ private List<TagHelperDescriptor> CreateEventHandlerTagHelpers(List<EventHandler
                 {
                     rule.TagName = "*";
 
+                    var entry1 = entry;
```
Comment (disklosr) [inline src/Components/src/Microsoft.AspNetCore.Components.Razor.Extensions/EventHandlerTagHelperDescriptorProvider.cs]: Is this declaration needed?

Comment (bytedreamer) [reply src/Components/src/Microsoft.AspNetCore.Components.Razor.Extensions/EventHandlerTagHelperDescriptorProvider.cs]: No it is not.

These files were deleted earlier today. Waiting to know if they were migrated to a new project.

**Comment by rynowak** [comment]
> No issues, did some refactoring while learning how the code works.

Then I'm not sure there's a reason we'd accept this PR. Please refer to our contribution guidelines [here](https://github.com/aspnet/AspNetCore/blob/master/CONTRIBUTING.md), and in particular this linked article: https://tirania.org/blog/archive/2010/Dec-31.html

If you think something is genuinely messy, and you can improve it, I think it would be a good idea to open an issue first next time to discuss.

We're definitely interested in your help fixing bugs or correcting things that are wrong or inefficient. Minor things like adding docs, and fixing spellign mistakes are welcome as well.


> These files were deleted earlier today. Waiting to know if they were migrated to a new project.

The new home for all of this is https://github.com/aspnet/AspNetCore-Tooling/blob/master/src/Razor/src/Microsoft.CodeAnalysis.Razor/EventHandlerTagHelperDescriptorProvider.cs

Sorry for the confusion, the compiler living this is repo was temporary. At the same time as we have been consolidating repos, we've also been adding support for components to the mainline Razor compiler. I threw the *big switch* last night, and we're now building on running on the official Razor bits rather than something I cobbled together 😆

**Comment by rynowak** [comment]
> Do you mind if I spend some time reducing the size of the methods of these providers? 

I'm really not sure what that means, so I don't think it's valuable.

I think I'd agree pretty wholeheartedly with what's mentioned by miguel in the  first two paragraphs of the article I linked. I don't really want to spend time negotiating over stylistic changes. If you find actual problems that should be solved I'm open to it.  

Thanks for your understanding and looking forward to hearing from you again soon.

**Comment by rynowak** [comment]
> I do find it interesting there is a push back from making code more understandable with this Open Source project. 

Sorry if you're getting that impression, I think what I'm trying to say is much more subtle than that. There isn't a push back on making code more understandable, that is one of many things we care about. 

> No worries, it is tough for me to understand code without spending some time refactoring.

If every individual does this, what is the result?

**Comment by bytedreamer** [comment]
I spend much of my time untangling large balls of code mess. To be clear, I'm not saying that this is the case with this code. In fact, I wish most of the code I dealt with was as well organized as this project. 

> If every individual does this, what is the result?

It has become habit for me to spend some time cleaning code and adding unit tests before proceeding. Gets me into the head of the previous developer to discover their intent. My hope is to reduce the ramp up time for the next developer.

This is a good discussion, I'm very much interested in how to keep existing code bases well maintained.  I'd imagine that maintaining an open source project presents different challenges from a localized closed source project.

**Pull request closed without merging**